Source/WebCore/bindings/js/JSPushNotificationEventCustom.cpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "JSPushNotificationEvent.h"
+
+#if ENABLE(DECLARATIVE_WEB_PUSH)
+
+#include "JSDOMWrapper.h"
+#include "PushNotificationEvent.h"
+
+namespace WebCore {
+
+JSC::JSValue JSPushNotificationEvent::proposedAppBadge(JSC::JSGlobalObject&) const
+{
+    // FIXME: This custom getter implements the following IDL intention:
+    // readonly attribute unsigned long long? proposedAppBadge;
```

Comment:
Are you sure this isn't supported? I thought this worked, as long as the implementation returned a `std::optional<unsigned long>`.

Reply:
I just checked and this is totally supported. We have plenty of examples in the existing code base where this is working. We don't need custom bindings.

Reply:
When I figured tried, it absolutely failed compiling miserably.

I *do* see other users of it, so I'm trying again.

Reply:
Ping me on Slack if you get into trouble with the generated bindings. Hopefully I can help. We should definitely be able to make this work without custom bindings.